channels/serializers.py

```diff
@@ -121,11 +122,23 @@ class PostSerializer(serializers.Serializer):
     created = serializers.SerializerMethodField()
     num_comments = serializers.IntegerField(read_only=True)
     channel_name = serializers.SerializerMethodField()
+    profile_image = serializers.SerializerMethodField()
 
     def get_url(self, instance):
         """Returns a url or null depending on if it's a self post"""
         return instance.url if not instance.is_self else None
 
+    def get_profile_image(self, instance):
+        """Find the Profile for the comment author"""
+        if instance.author is None:
+            return default_profile_image
+
+        profile = User.objects.get(username=instance.author.name).profile
+        if profile.image_small:
+            return profile.image_small
+        else:
+            return default_profile_image
```

Comment:
Can you add a test for this case?

Reply:
I believe the `test_get_post` test (in `channels/views_test.py`) covers this case, unless I'm mistaken

Reply:
It looks like `create_profile_and_login` uses `ProfileFactory` to create the profile, which fills in values for the images here, so `profile.image_small` would always be truthy

Reply:
yes, but there's a test where `post.author` is `None`, which tests that path.

Reply:
oh nvm yeah I do need to add another one :woman_facepalming: